src/lib/Libifl/pbsD_submit.c

```diff
@@ -218,23 +221,37 @@ __pbs_submit(int c, struct attropl  *attrib, char *script, char *destination, ch
 			extend = EXTEND_OPT_IMPLICIT_COMMIT;
 	}	
 
-	c = random_svr_conn;	
-	if (msvr && destination) {
+	if (msvr && IS_EMPTY(dest)) {
 		/* Reached here means job is submitted to non default queue */
-		int start;
 
 		/* Since this could be a reservation queue and reservation queues are not shared,
 		 * we try to find out which server this queue resides on
 		 */
-		if ((start = get_obj_location_hint(destination, MGR_OBJ_RESV)) != -1)
-			c = svr_conns[start]->sd;
+		if ((start = get_obj_location_hint(dest, MGR_OBJ_RESV)) == -1)
+			start = 0;
 	}
 	
-	/* Queue job with null string for job id */
-	return_jobid = PBSD_queuejob(c, "", destination, attrib, extend, PROT_TCP, NULL, &commit_done);
-	if (return_jobid == NULL)
+	/* Queue job with null string for job id
+	* attempt again with other instances if we get a queued limit error.
+	*/
+	for (i = start, ct = 0; ct < nsvr; i = (i + 1) % nsvr, ct++) {
+
+		if (!svr_conns[i] || svr_conns[i]->state != SVR_CONN_STATE_UP) {
+			rc = PBSE_NOSERVER;
+			continue;
+		}
+
+		c = svr_conns[i]->sd;
+		return_jobid = PBSD_queuejob(c, "", dest, attrib, extend, PROT_TCP, NULL, &commit_done);
+		if (return_jobid)
+			break;
+		else if (IS_EMPTY(dest) || (pbs_errno != PBSE_MAXQUED && pbs_errno != PBSE_ENTLIMCT && pbs_errno != PBSE_ENTLIMRESC))
+				goto error;
```

Comment:
Not sure I understand this. If 'dest' was given, we'll not try other servers? Why? 'dest' is just a queue name, which can be shared for all servers.

Also, I think we should try all servers regardless of what the error was. Is there any case where it'd be bad if we tried all servers without checking for error?